### PR TITLE
[W4.11e][W17-3]Ooi Heng Seng

### PR DESCRIPTION
**Pull request by jeffreyooi**

### CODE REVIEW

File: test/java/seedu/addressbook/common/UtilsTest.java [hunk 1]
```diff
@@ -43,4 +53,12 @@ private void assertAreUnique(Object... objects) {
...
         assertFalse(Utils.elementsAreUnique(Arrays.asList(objects)));
     }
+
+    private void isAnyNull(Object... objects) {
+        assertTrue(Utils.isAnyNull(objects));
+    }
+
+    private void isAllNonNull(Object... objects) {
+        assertFalse(Utils.isAnyNull(objects));
+    }
```
Comment (parvathy-sudhir-pillai) [inline test/java/seedu/addressbook/common/UtilsTest.java:63]: It's good to follow the style used by the code nearby 👍
Having said that, it is doubtful if extracting these two methods add any value.
You could use the assertFalse and assertTrue inline as there are no conversions needed unlike in the elementsAreUnique method.

File: test/java/seedu/addressbook/common/UtilsTest.java [hunk 2]
```diff
@@ -34,6 +34,16 @@ public void elementsAreUnique() throws Exception {
         assertNotUnique(null, 1, Integer.valueOf(1));
         assertNotUnique(null, null);
         assertNotUnique(null, "a", "b", null);
+
+        isAnyNull("abc", 555, null);
```
Comment (parvathy-sudhir-pillai) [inline test/java/seedu/addressbook/common/UtilsTest.java:38]: Test cases for isAnyNull() method from Utils.java, should be in separate method in UtilsTest.java. You may follow this practice that for every method being tested there should be a corresponding method of the same name in the test.java file. The problem with your current version is that you do not separate the method which tests for elementsAreUnique() and isAnyNull(), which is not a good practice. You should be more careful about separation of concerns for each test method.

File: test/java/seedu/addressbook/common/UtilsTest.java [hunk 3]
```diff
@@ -34,6 +34,16 @@ public void elementsAreUnique() throws Exception {
...
+        isAnyNull((Object) null);
+        isAnyNull(Integer.valueOf(-123), 1, null);
+        isAnyNull(null, null, null);
+
```
Comment (parvathy-sudhir-pillai) [inline test/java/seedu/addressbook/common/UtilsTest.java:42]: Could add a few comments as to what is being tested in these separate logical units of code in this block.

File: test/java/seedu/addressbook/common/UtilsTest.java [hunk 4]
```diff
@@ -34,6 +34,16 @@ public void elementsAreUnique() throws Exception {
         assertNotUnique(null, 1, Integer.valueOf(1));
         assertNotUnique(null, null);
         assertNotUnique(null, "a", "b", null);
+
+        isAnyNull("abc", 555, null);
+        isAnyNull((Object) null);
+        isAnyNull(Integer.valueOf(-123), 1, null);
+        isAnyNull(null, null, null);
```
Comment (parvathy-sudhir-pillai) [inline test/java/seedu/addressbook/common/UtilsTest.java:41]: Could add a test case for empty list.

**Review by parvathy-sudhir-pillai — COMMENTED** [review]
Good effort @jeffreyooi. Please close this PR after reading the review comments.
